### PR TITLE
RATEST-308:NEW E2E Test : Check XSS Vulnerability Workflow

### PR DESCRIPTION
**Pull request by sherrif10**
https://issues.openmrs.org/browse/RATEST-308

### CODE REVIEW

File: qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/XssOnPhoneNumberFieldSteps.java
Comment (mherman22) [inline qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/XssOnPhoneNumberFieldSteps.java:26]:
```suggestion
    public void setUp() throws Exception {
```

File: qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/XssOnPhoneNumberFieldSteps.java
Comment (mherman22) [inline qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/XssOnPhoneNumberFieldSteps.java:33]:
```suggestion
    public void tearDown() throws Exception {
```

File: qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/XssOnPhoneNumberFieldSteps.java
Comment (mherman22) [inline qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/XssOnPhoneNumberFieldSteps.java:63]: Some indention correction needed here

**Review by mherman22 — CHANGES_REQUESTED** [review]

**Comment by sherrif10** [comment]
@kdaud  Incase you find time. Feel free to look through the pull request. And probably other pull requests that are waiting for approval then i can do the needfull.

**Comment by kdaud** [comment]
@sherrif10 have put a comment on the ticket.

File: qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/XssOnPhoneNumberFieldSteps.java
Comment (kdaud) [inline qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/XssOnPhoneNumberFieldSteps.java:34]: how is the test quitting  the run?

Comment (sherrif10) [reply qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/XssOnPhoneNumberFieldSteps.java:34]: As you can see the method used above. After all the functionalities are done., the test is supposed to delete that patient and quits the browser.

Comment (kdaud) [reply qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/XssOnPhoneNumberFieldSteps.java:34]: could you point to the method that makes the test to quit the run?

Comment (sherrif10) [reply qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/XssOnPhoneNumberFieldSteps.java:34]: > could you refer to the method that makes the test to quit the run?

@kdaud  Its not by default that the test is supposed to quit using **quit()** method. the same exact method . delete() quits the browser.

Comment (kdaud) [reply qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/XssOnPhoneNumberFieldSteps.java:34]: where is the delete() function that you are referring to?

Comment (sherrif10) [reply qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/XssOnPhoneNumberFieldSteps.java:34]: Resolved thanks.

Comment (sherrif10) [reply qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/XssOnPhoneNumberFieldSteps.java:34]: Responded to your question with a commit.

File: qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/XssOnPhoneNumberFieldSteps.java
Comment (kdaud) [inline qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/XssOnPhoneNumberFieldSteps.java:16]: cross check the order of the imports in the existing files and align.

Comment (sherrif10) [reply qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/XssOnPhoneNumberFieldSteps.java:10]: Resolved.

Comment (sherrif10) [reply qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/XssOnPhoneNumberFieldSteps.java:10]: Let me know if we have more reviews such that i can file them quickly.

File: qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/XssOnPhoneNumberFieldSteps.java
Comment (kdaud) [inline qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/XssOnPhoneNumberFieldSteps.java:22]: do you need to declare the variable patientDashboardPage?

File: qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/XssOnPhoneNumberFieldSteps.java
Comment (sherrif10) [inline qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/XssOnPhoneNumberFieldSteps.java:105]: @kdaud am implementing it here

Comment (kdaud) [reply qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/XssOnPhoneNumberFieldSteps.java:105]: why not take advantage of inheritance?

Comment (sherrif10) [reply qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/XssOnPhoneNumberFieldSteps.java]: Resolved.

**Comment by sherrif10** [comment]
cc @kdaud  more reviews , approval required :smile_cat:

File: qaframework-bdd-tests/src/test/resources/features/refapp-2.x/016-XssVulnerability/xssVulnerability.feature
Comment (kdaud) [inline qaframework-bdd-tests/src/test/resources/features/refapp-2.x/016-XssVulnerability/xssVulnerability.feature:19]:
```suggestion
    And User clears the phone Number input area
```

File: qaframework-bdd-tests/src/test/resources/features/refapp-2.x/016-XssVulnerability/xssVulnerability.feature
Comment (kdaud) [inline qaframework-bdd-tests/src/test/resources/features/refapp-2.x/016-XssVulnerability/xssVulnerability.feature:18]: any reason for adding an empty line here?

Comment (sherrif10) [reply qaframework-bdd-tests/src/test/resources/features/refapp-2.x/016-XssVulnerability/xssVulnerability.feature:18]: The slight reason would be to clearly follow the scenarios properly since they are gona be compact.Another reason is that since the test clears the input text twice, i would love seeing where it clears again, so  i add another line for formality and clean code presumably.

Comment (kdaud) [reply qaframework-bdd-tests/src/test/resources/features/refapp-2.x/016-XssVulnerability/xssVulnerability.feature:18]: does it each step tells clearly what is being done?

Comment (sherrif10) [reply qaframework-bdd-tests/src/test/resources/features/refapp-2.x/016-XssVulnerability/xssVulnerability.feature]: Resolved thanks. Though it doesnt look morethan good . Remember these are human readable scenarios. i personally discourage passages without paragraphs.

Comment (kdaud) [reply qaframework-bdd-tests/src/test/resources/features/refapp-2.x/016-XssVulnerability/xssVulnerability.feature]: why do you think there should be an empty line? And when should an empty line be added within the feature file?

File: qaframework-bdd-tests/src/test/resources/features/refapp-2.x/016-XssVulnerability/xssVulnerability.feature
Comment (sherrif10) [reply qaframework-bdd-tests/src/test/resources/features/refapp-2.x/016-XssVulnerability/xssVulnerability.feature]: The idea is to have clean , readable scenario. It however depends on the fact of functionality

**Comment by sherrif10** [comment]
This should work if [this ](https://github.com/openmrs/openmrs-contrib-qaframework/pull/340)gets finished

**Comment by sherrif10** [comment]
Hello @kdaud  this ticket is one of the ticket we would love merging incase everything is done.

File: qaframework-bdd-tests/src/test/resources/features/refapp-2.x/015-reports/reports.feature
Comment (kdaud) [inline qaframework-bdd-tests/src/test/resources/features/refapp-2.x/015-reports/reports.feature:17]: is this change part of the ticket requirement?

Comment (sherrif10) [reply qaframework-bdd-tests/src/test/resources/features/refapp-2.x/015-reports/reports.feature:17]: Yes it is temporarily part of the requirement as am waiting for report to be fixed, then will update it accordingly.

File: qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/RunTest.java
Comment (sherrif10) [inline qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/RunTest.java:34]: Hello @kdaud  Looking forward for reviews, recommendations if everything looks fine. Merge. thanks

**Comment by kdaud** [comment]
@sherrif10 our platform tests are failing build plans as result of the recent commits, would be a good plan to fix them first.

**Comment by sherrif10** [comment]
> platform tests are failing build plans as result of the recent commits,

Am not sure if its because of the recent commits. However let me look through them

File: qaframework-bdd-tests/src/test/resources/features/refapp-2.x/016-XssVulnerability/xssVulnerability.feature
Comment (kdaud) [inline qaframework-bdd-tests/src/test/resources/features/refapp-2.x/016-XssVulnerability/xssVulnerability.feature:23]: how about lower casing all the user steps to have uniformity in all our feature file stories?
```suggestion
    Then user validate the collect number to be entered into the input area
```

**Comment by sherrif10** [comment]
cc @kdaud is this yet ripe.

File: qaframework-bdd-tests/src/test/resources/features/refapp-2.x/016-XssVulnerability/xssVulnerability.feature
Comment (sherrif10) [reply qaframework-bdd-tests/src/test/resources/features/refapp-2.x/016-XssVulnerability/xssVulnerability.feature]: Resolved thnks

File: qaframework-bdd-tests/src/test/resources/features/refapp-2.x/016-XssVulnerability/xssVulnerability.feature
Comment (kdaud) [inline qaframework-bdd-tests/src/test/resources/features/refapp-2.x/016-XssVulnerability/xssVulnerability.feature:10]: I think we can have these two steps under background. something like

```js
Background:
    Given user goes to active visit search page
    When user searches for patient using patient identifier
    Then the system goes to patient dashboard of last Active visit

Comment (sherrif10) [reply qaframework-bdd-tests/src/test/resources/features/refapp-2.x/016-XssVulnerability/xssVulnerability.feature:10]: I don't see any reason why we are adding  it in background

Comment (kdaud) [reply qaframework-bdd-tests/src/test/resources/features/refapp-2.x/016-XssVulnerability/xssVulnerability.feature:10]: the scenario should show the actual steps involved in checking xss so those two steps suits to be within the background part.

Comment (sherrif10) [reply qaframework-bdd-tests/src/test/resources/features/refapp-2.x/016-XssVulnerability/xssVulnerability.feature:10]: I really don't see any strong reason. What am seeing here is initiateLogin. All other scenario fits best in part of scenarios not in background. Let's try to use background part appropriately. Adding. Scenarios in background simply means that there are some scenarios you ensure to be run first.


On my scenario.  I expect initiateLogin to first run after every scenario. And this have been achieved with in before method. So it won't look a bit practical.

Comment (kdaud) [reply qaframework-bdd-tests/src/test/resources/features/refapp-2.x/016-XssVulnerability/xssVulnerability.feature:10]:
```suggestion
    Then the system loads patient dashboard page of last Active visit
```

Comment (sherrif10) [reply qaframework-bdd-tests/src/test/resources/features/refapp-2.x/016-XssVulnerability/xssVulnerability.feature]: @kdaud  does the language looks fine. i mean the scenario

File: qaframework-bdd-tests/src/test/resources/features/refapp-2.x/016-XssVulnerability/xssVulnerability.feature
Comment (kdaud) [inline qaframework-bdd-tests/src/test/resources/features/refapp-2.x/016-XssVulnerability/xssVulnerability.feature:10]:
```suggestion
    Then system loads  patient dashboard page of last Active visit
```

File: qaframework-bdd-tests/src/test/resources/features/refapp-2.x/016-XssVulnerability/xssVulnerability.feature
Comment (kdaud) [inline qaframework-bdd-tests/src/test/resources/features/refapp-2.x/016-XssVulnerability/xssVulnerability.feature:11]:
```suggestion
    And user clicks on show contact link
```

Comment (sherrif10) [reply qaframework-bdd-tests/src/test/resources/features/refapp-2.x/016-XssVulnerability/xssVulnerability.feature]: Resolved thanks.

**Comment by sherrif10** [comment]
@kdaud  this kind of breakages must be brought about a change onto test.local.properties.our tests are not reliable on that file especiallly on gitub actions.

**Comment by kdaud** [comment]
The recent merges on master let the test breakage and am have made `pull/354 `to fix

**Comment by sherrif10** [comment]
Hi @kdaud  this is what happen on github actions.hopefully your commit will resurrect these failures

**Comment by sherrif10** [comment]
cc @kdaud  Looking forward for approval.

File: qaframework-bdd-tests/src/test/resources/features/refapp-2.x/015-reports/reports.feature
Comment (kdaud) [inline qaframework-bdd-tests/src/test/resources/features/refapp-2.x/015-reports/reports.feature:17]: is this change part of the ticket requirement?

File: qaframework-bdd-tests/src/test/resources/org/openmrs/uitestframework/test-local.properties
Comment (kdaud) [inline qaframework-bdd-tests/src/test/resources/org/openmrs/uitestframework/test-local.properties:10]: Any reason for this change?

File: qaframework-bdd-tests/src/test/resources/features/refapp-2.x/016-XssVulnerability/xssVulnerability.feature
Comment (kdaud) [reply qaframework-bdd-tests/src/test/resources/features/refapp-2.x/016-XssVulnerability/xssVulnerability.feature:10]: did you see the comment [above](https://github.com/openmrs/openmrs-contrib-qaframework/pull/327/files#r802017014)?

File: qaframework-bdd-tests/src/test/resources/org/openmrs/uitestframework/test-local.properties
Comment (sherrif10) [reply qaframework-bdd-tests/src/test/resources/org/openmrs/uitestframework/test-local.properties:10]: No change here checkout this line https://github.com/openmrs/openmrs-contrib-qaframework/blob/master/qaframework-bdd-tests/src/test/resources/org/openmrs/uitestframework/test-local.properties#L10

File: qaframework-bdd-tests/src/test/resources/features/refapp-2.x/015-reports/reports.feature
Comment (sherrif10) [reply qaframework-bdd-tests/src/test/resources/features/refapp-2.x/015-reports/reports.feature:17]: checkout this line on master branch.   these were brought about updating my branch to master  https://github.com/openmrs/openmrs-contrib-qaframework/blob/master/qaframework-bdd-tests/src/test/resources/features/refapp-2.x/015-reports/reports.feature#L17

File: qaframework-bdd-tests/src/test/resources/features/refapp-2.x/016-XssVulnerability/xssVulnerability.feature
Comment (sherrif10) [reply qaframework-bdd-tests/src/test/resources/features/refapp-2.x/016-XssVulnerability/xssVulnerability.feature:10]: Resolved thanks.

File: qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/XssOnPhoneNumberFieldSteps.java
Comment (kdaud) [inline qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/XssOnPhoneNumberFieldSteps.java:37]:
```suggestion
        new TestData.TestVisit(patient.uuid, TestData.getAVisitType(),  getLocationUuid(homePage)).create();
```

Comment (sherrif10) [reply qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/XssOnPhoneNumberFieldSteps.java:37]: Hey @kdaud  what is the difference here according to your suggestion above.

File: qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/XssOnPhoneNumberFieldSteps.java
Comment (kdaud) [inline qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/XssOnPhoneNumberFieldSteps.java:37]: can't we have this on the same line?

File: qaframework-bdd-tests/src/test/resources/org/openmrs/uitestframework/test-local.properties
Comment (kdaud) [inline qaframework-bdd-tests/src/test/resources/org/openmrs/uitestframework/test-local.properties:1]: any reason for this change?

File: qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/XssOnPhoneNumberFieldSteps.java
Comment (kdaud) [reply qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/XssOnPhoneNumberFieldSteps.java]: checkout my recent comments.

Comment (sherrif10) [reply qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/XssOnPhoneNumberFieldSteps.java]: Resolved.

File: qaframework-bdd-tests/src/test/resources/org/openmrs/uitestframework/test-local.properties
Comment (sherrif10) [reply qaframework-bdd-tests/src/test/resources/org/openmrs/uitestframework/test-local.properties]: Resolved

File: qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/XssOnPhoneNumberFieldSteps.java
Comment (sherrif10) [reply qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/XssOnPhoneNumberFieldSteps.java]: Resolved

File: qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/XssOnPhoneNumberFieldSteps.java
Comment (kdaud) [inline qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/XssOnPhoneNumberFieldSteps.java:36]:
```suggestion
        new TestData.TestVisit(patient.uuid, TestData.getAVisitType(), getLocationUuid(homePage)).create();
```

Comment (sherrif10) [reply qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/XssOnPhoneNumberFieldSteps.java]: Resolved thanks.

File: qaframework-bdd-tests/src/test/resources/features/refapp-2.x/016-XssVulnerability/xssVulnerability.feature
Comment (kdaud) [inline qaframework-bdd-tests/src/test/resources/features/refapp-2.x/016-XssVulnerability/xssVulnerability.feature:12]:
```suggestion
    And user clicks on edit contact info
```

Comment (sherrif10) [reply qaframework-bdd-tests/src/test/resources/features/refapp-2.x/016-XssVulnerability/xssVulnerability.feature]: Resolved thanks.

File: qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/XssOnPhoneNumberFieldSteps.java
Comment (kdaud) [inline qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/XssOnPhoneNumberFieldSteps.java:36]:
```suggestion
        new TestData.TestVisit(patient.uuid, TestData.getAVisitType(), getLocationUuid(homePage)).create();
```

File: qaframework-bdd-tests/src/test/resources/features/refapp-2.x/016-XssVulnerability/xssVulnerability.feature
Comment (kdaud) [inline qaframework-bdd-tests/src/test/resources/features/refapp-2.x/016-XssVulnerability/xssVulnerability.feature:12]:
```suggestion
    And user clicks on edit contact info
```

File: qaframework-bdd-tests/src/test/resources/features/refapp-2.x/016-XssVulnerability/xssVulnerability.feature
Comment (kdaud) [inline qaframework-bdd-tests/src/test/resources/features/refapp-2.x/016-XssVulnerability/xssVulnerability.feature:10]:
```suggestion
    Then system loads  patient dashboard page of last active visit
```

File: qaframework-bdd-tests/src/test/resources/features/refapp-2.x/016-XssVulnerability/xssVulnerability.feature
Comment (kdaud) [inline qaframework-bdd-tests/src/test/resources/features/refapp-2.x/016-XssVulnerability/xssVulnerability.feature:16]:
```suggestion
    And user clicks on confirm edit button
```

Comment (sherrif10) [reply qaframework-bdd-tests/src/test/resources/features/refapp-2.x/016-XssVulnerability/xssVulnerability.feature]: Resolved

**Pull request closed without merging**
